shared/src/commonMain/kotlin/org/hyperskill/app/streak_recovery/presentation/StreakRecoveryActionDispatcher.kt

```diff
@@ -0,0 +1,81 @@
+package org.hyperskill.app.streak_recovery.presentation
+
+import org.hyperskill.app.analytic.domain.interactor.AnalyticInteractor
+import org.hyperskill.app.core.presentation.ActionDispatcherOptions
+import org.hyperskill.app.profile.domain.interactor.ProfileInteractor
+import org.hyperskill.app.streak_recovery.presentation.StreakRecoveryFeature.Action
+import org.hyperskill.app.streak_recovery.presentation.StreakRecoveryFeature.Message
+import org.hyperskill.app.streaks.domain.flow.StreakFlow
+import org.hyperskill.app.streaks.domain.interactor.StreaksInteractor
+import ru.nobird.app.presentation.redux.dispatcher.CoroutineActionDispatcher
+
+class StreakRecoveryActionDispatcher(
+    config: ActionDispatcherOptions,
+    private val profileInteractor: ProfileInteractor,
+    private val streaksInteractor: StreaksInteractor,
+    private val analyticInteractor: AnalyticInteractor,
+    private val streakFlow: StreakFlow
+) : CoroutineActionDispatcher<Action, Message>(config.createConfig()) {
+    override suspend fun doSuspendableAction(action: Action) {
+        when (action) {
+            StreakRecoveryFeature.InternalAction.FetchStreak -> {
+                // TODO: replace with current profile state repository when ALTAPPS-682 will be ready
```

Comment:
Now it's possible to use the current profile state repository, needs to merge `develop`